test/unit/io_managers_test.py

```diff
@@ -277,6 +280,33 @@ def test_error_when_handling_view_without_metadata(fake_pudl_sqlite_io_manager_f
         fake_pudl_sqlite_io_manager_fixture.handle_output(output_context, sql_stmt)
 
 
+def test_empty_read_fails(fake_pudl_sqlite_io_manager_fixture):
+    """Reading empty table fails."""
+    with pytest.raises(AssertionError):
+        context = build_input_context(asset_key=AssetKey("artist"))
+        fake_pudl_sqlite_io_manager_fixture.load_input(context)
+
+
+def test_replace_on_insert(fake_pudl_sqlite_io_manager_fixture):
+    """Tests that two runs of the same asset overwrite existing contents."""
+    artist_df = pd.DataFrame({"artistid": [1], "artistname": ["Co-op Mop"]})
+    output_context = build_output_context(asset_key=AssetKey("artist"))
+    input_context = build_input_context(asset_key=AssetKey("artist"))
+
+    # Write then read.
+    fake_pudl_sqlite_io_manager_fixture.handle_output(output_context, artist_df)
+    read_df = fake_pudl_sqlite_io_manager_fixture.load_input(input_context)
+    pd.testing.assert_frame_equal(artist_df, read_df, check_dtype=False)
+    # check_dtype=False, because int64 != Int64. /o\
```

Comment:
Should we be getting matching dtypes here, if we're using the `pudl_sqlite_io_manager` since it uses `enforce_schema(df)` before writing and after reading? Is that having no effect because there's not actually any `Resource` defined for the tables being written/read with specific data types? I don't really understand the how or why of mocking.

Reply:
It is applying `enforce_schema` because earlier in this file the `Package` is created with the schema, so I think it's the original `DataFrame` where the types are off. We could set those explicitly so we can check dtypes, but if that would be better.